tracee-rules/signature.go

```diff
@@ -45,27 +45,28 @@ func getSignatures(rulesDir string, rules []string) ([]types.Signature, error) {
 }
 
 func findGoSigs(dir string) ([]types.Signature, error) {
-	files, err := ioutil.ReadDir(dir)
-	if err != nil {
-		return nil, fmt.Errorf("error reading plugins directory %s: %v", dir, err)
-	}
 	var res []types.Signature
-	for _, file := range files {
-		if filepath.Ext(file.Name()) != ".so" {
-			continue
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("error opening plugin %s: %v", info.Name(), err)
```

Comment:
to my understanding this will cause the walker to abort. isn't it better to return SkipDir (and log it) so that other files might be still processed? same goes to all other errors returned from this func

Reply:
didn't you mean to SkipDir here too?

Reply:
also, log the error like in other cases